python/src/equistore/operations/_dispatch.py

```diff
@@ -129,3 +129,15 @@ def vstack(arrays):
         return torch.vstack(arrays)
     else:
         raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def index_add(array, shape0, index):
```

Comment:
This function feels like it is doing the job of two functions: `zero_like` and then `index_add`. How about splitting it in two?